src/Client.ts

```diff
@@ -351,6 +351,17 @@ export class Client extends EventEmitter {
         return await this.req('GET', '/sync/unreads');
     }
 
+    /**
+     * 
+     * @param text Presence text which will be set
+     * @param presence Presence type which will be displayed
+     * @returns any
+     */
+
+    async setPresence(text: string, presence: any = 'Online') {
+        return this.req('PATCH', '/users/@me' as '/users/id', { status: { text, presence } })
```

Comment:
`async` keyword should be removed here to fix types and to correctly block execution further up.